main.cpp

```diff
@@ -20,29 +10,37 @@ vector<string> split(string path) {
 		directories.push_back(element);
 		path = path.substr(cut + 1, path.size());
 	}
-
 	return directories;
 }
 
 int main() {
-	cout << "Welcome in the terminal. Press 'q' to exit." << endl;
+	cout << "Please add a valid file name to load filesystem from. " << endl;
+	cout << "Welcome in the terminal. Write exit' to exit." << endl;
 	string CurrentFolder = "root";
 	Dictionary d;
 	string parancs;
-
-	while (parancs != "q") {
+	string dirname;
+	string fsname;
+	cin >> fsname;
+	while (d.validcommand(fsname)) {
+		cout << "Not valid name. Please add a valid file name to load filesystem from. " << endl;
+		cin >> fsname;
+		if (fsname == "exit") break;
+	}
+	d.loadFromFile(fsname);
+	while (parancs != "exit") {
 		string autotext = "C:/" + CurrentFolder + ">";
 		cout << autotext;
 		cin >> parancs;
-		if (!validcommand(parancs)) {
+		if (parancs == "exit") break;
```

Comment:
na itt ez az if kaosz ami lentebb van mar atlathatatlan, tessek rendbeszedni :-) 
Egyreszt a parancsok kozotti szetvalasztas legyen egy levelen, masreszt ilyen /-enkenti parzolgatast ne a main-be tegyunk mar bele.

Reply:
Javítottam, rerequesteltem, a parzolast száműztem functionbe. :)